mlflow/utils/autologging_utils/logging_and_warnings.py

```diff
@@ -94,11 +95,10 @@ def _modify_patch_state_if_necessary(self):
         """
         with self._state_lock:
             if self._should_patch_showwarning() and not self._did_patch_showwarning:
-                self._original_showwarning = warnings.showwarning
```

Comment:
This line assumes that `warnings.showwarning` is the original `warnings.showwarning`, but if it's `self._patched_showwarning`, infinite recursion occurs.